platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -168,58 +249,102 @@ define(
             this.$scope.headerHeight = headerHeight;
             this.$scope.rowHeight = rowHeight;
             this.$scope.totalHeight = overallHeight;
-
-            this.$scope.visibleRows = this.$scope.displayRows.slice(0, this.maxDisplayRows).map(function(row, i) {
-                return {
-                    rowIndex: i,
-                    offsetY: (i * self.$scope.rowHeight) + self.$scope.headerHeight,
-                    contents: row
-                };
-            });
-
+            this.setVisibleRows();
             this.$scope.overrideRowPositioning = true;
         };
 
         /**
-         * Returns a new array which is a result of applying the sort
-         * criteria defined in $scope.
-         *
-         * Does not modify the array that was passed in.
+         * @private
          */
-        MCTTableController.prototype.sortRows = function(rowsToSort) {
-            /**
-             * Compare two variables, returning a number that represents
-             * which is larger.  Similar to the default array sort
-             * comparator, but does not coerce all values to string before
-             * conversion.  Strings are lowercased before comparison.
-             */
-            function genericComparator(a, b) {
-                if (typeof a === "string" && typeof b === "string") {
-                    a = a.toLowerCase();
-                    b = b.toLowerCase();
+        MCTTableController.prototype.insertSorted = function(array, element) {
+            var index = -1,
+                self = this,
+                sortKey = this.$scope.sortColumn;
+
+            function binarySearch(searchArray, searchElement, min, max){
```

Comment:
Nice binary search!  Props for implementing the non-overflowing average calculation that would have broken our table display when trying to display 9,007,199,254,740,992 rows (http://googleresearch.blogspot.com/2006/06/extra-extra-read-all-about-it-nearly.html).

I imagine at some point we'll pull in a nice functional library like lodash and use _.sortedIndex, but until then this is A+.